### PR TITLE
Feat/premade templates

### LINKED ISSUE #194: Improve the publishing pages flow for our content team
### Description:

As discussed on [this issue ](https://github.com/Codeinwp/templates-patterns-collection/issues/164), it would help to have a way of automatically generating the Screenshots when we (Themeisle admin) uploads page in the TPC (attached the pages link I am referring to)
Screenshot 2022-12-23 at 14 35 19

___

I am open to any news or thoughts on how we can approach this?
Essentially the challenge is the fact that (we, as Themeisle admins) to add a Demo pages in Templates cloud, need to:

- take a screenshot of the demo and adjust it's size
- go on that page in edit mode
- add the entries like site slug & modified screenshot url
- and this takes some time.

The ideal solution would be that we have a license/account just for upload, and that account only need to go on the page in edit mode, add the site slug - press publish, and the screenshot will be generated -> page published.

### Alternatives:

Do nothing

### PR DESCRIPTION
**Pull request by preda-bogdan**
### Summary

Changes for premade templates with automatic screenshot generation.

### Test instructions

Testing instructions are provided here: https://github.com/Codeinwp/templates-cloud/pull/60

Closes #194.

### CODE REVIEW

File: elementor/src/components/export.js [hunk 1]
```diff
@@ -101,31 +102,47 @@ const Export = ( { updateCurrentTab } ) => {
...
 			if ( r.success ) {
-				elementor.notifications.showToast( {
-					message: ! isPublished
-						? window.tiTpc.exporter.templatePublished
-						: window.tiTpc.exporter.templateUnpublished,
+				await getTemplate( templateID ).then( ( results ) => {
+					console.log( templateID );
+					console.log( results );
+					console.log( results.template_id );
+					console.log( results.template_thumbnail );
```
Comment (abaicus) [inline elementor/src/components/export.js:111]: We can probably drop the console logs here.

File: editor/src/extension.js [hunk 2]
```diff
@@ -321,18 +321,24 @@ const Exporter = () => {
...
+								published
+									? __( 'Template Unpublished.', 'templates-patterns-collection' )
+									: __( 'Template Published.', 'templates-patterns-collection' ),
+								{
+									type: 'snackbar',
+								}
+							);
 						}
-					);
+					} );
```
Comment (abaicus) [inline editor/src/extension.js:340]: We can add a catch to this to ensure that stuff still works if the request fails. We should probably do the same for the `publishTemplate` call above.

**Review by abaicus — CHANGES_REQUESTED** [review]

**Comment by preda-bogdan** [comment]
Thank you @abaicus, I forgot about those, I wrapped the whole call with a try-catch to handle possible fetch errors.

File: editor/src/extension.js [hunk 3]
```diff
@@ -428,6 +438,9 @@ const Exporter = () => {
 						<TextControl
 							label={ __( 'Screenshot URL' ) }
 							value={ screenshotURL }
+							help={ __(
```
Comment (HardeepAsrani) [inline editor/src/extension.js:441]: In few places, we use the i18n functions without the text-domain.

File: elementor/src/components/export.js [hunk 4]
```diff
@@ -94,38 +95,55 @@ const Export = ( { updateCurrentTab } ) => {
...
+							} );
+
+							window.tiTpc.postModel.save();
+						}
+					} );
+				}
+			} );
+		} catch ( error ) {
+			elementor.notifications.showToast( {
+				message: 'Something happened when publishing the template.',
```
Comment (HardeepAsrani) [inline elementor/src/components/export.js:133]: Similarly we don't use any localization function here.

**Review by HardeepAsrani — APPROVED** [review]
Looks good to go overall, apart from things Andrei pointed out, there are few i18n issues here and there that should be corrected.

**Pull request merged**